aiida/tools/importexport/common/zip_folder.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Export a zip-file."""
+# pylint: disable=missing-docstring,redefined-builtin
+import io
+import os
+import zipfile
+
+
+class MyWritingZipFile:
```

Comment:
Not sure this name is self-explanatory. Could you add some docstrings to explain what this does?